src/cli.ts

```
@@ -29,7 +29,15 @@ export class CLI {
         let gradleCommand = this.getGradleCommand();
         if (!gaugeCommand || gaugeCommand === '') return new CLI(gaugeCommand, {}, mvnCommand, gradleCommand);
         let gv = spawnSync(gaugeCommand, [GaugeCommands.Version, GaugeCommands.MachineReadable]);
-        return new CLI(gaugeCommand, JSON.parse(gv.stdout.toString()), mvnCommand, gradleCommand);
+        let gaugeVersionInfo;
+        try {
+            gaugeVersionInfo = JSON.parse(gv.stdout.toString());
+        } catch (e) {
+            window.showErrorMessage(`Unable to parse\n${gv.stdout.toString()}`);
```

Calling `window.showErrorMessage` will result in 2 separate errors being reported. I think we should report a single error. Also, the stacktrace is going to be noise since it will contain `JSON.parse`'s error stacktrace

---

Then `window.showErrorMessage('Unable to parse\n${gv.stdout.toString()}');` should be enough.

---

Seems like the fix should be "if you don't have any plugins installed, then the program should resolve in a way that doesn't throw an error"

---

@nathanBrenner Yes, you are right but that scenario will be handled by Gauge itself. Here we are just trying to be sure that the plugin activation does not fail silently.

---

I guess the errror message can include something in the lines of, `"Gauge expects a language runner to be installed. Error fetching plugin list from" + gv.stdout.toString()`

---

I.M.O vscode plugin activation should not be dependent on language runner being installed. As vscode plugin takes care of this by asking the user to install a relavent language plugin when a Gauge project is opened.

Though, I agree on changing the error message to somthing like `"Error fetching Gauge and plugins version information." + gv.stdout.toString()`

---

Fair enough, let's get the error message sorted.